### PR TITLE
EnsembleForward seeding, reseed pickles

### LINKED ISSUE #384: Initial Seed in ForwardEnsembleSampler
--------
Issue Description
--------
The initialSeed should be specifiable for the EnsembleSampler in a samplerInit block, but this was not possible due to the error checking.  Additionally, it ignores all sub-sampler initial seeds.

It would be nice to be able to set a seed for forward ensemble samplers.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or improvement?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed? *Improvement made*
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.) *Tested for in regression system*
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)? *unlikely to impact users*
- [x] 4. If the issue is a defect, does it impact the latest stable branch? If yes, is there any issue tagged with stable (create if needed)? *Not defect*
- [x] 5. If the issue is being closed without a merge request, has an explanation of why it is being closed been provided? *Will be closed with merge request*

Issue Close Review by @joshua-cogliati-inl on 2017-Sept-27

### PR DESCRIPTION
**Pull request by PaulTalbot-INL**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Closes #384 

##### What are the significant changes in functionality due to this change request?
 - Makes `<samplerInit>` work for the EnsembleForward sampler
 - Puts in place mechanics for pickledROMs setting the global RNG seed

Note: while the pickledROM can set seeds, the seed is ALWAYS reset by samplers in the initialization of MultiRun steps, so the pickle seed may be overwritten before it gets sampled.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code. *Done*
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts). *changes made*
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details. *Yes*
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.  *Passed before documentation only change, and automated document building passed*
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True. *Test updated*
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync. *No change*
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.  *Close checklist done*
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added? *No change*

As soon as the tests pass, the checklist passes. @joshua-cogliati-inl on 2017-Sept-27

### CODE REVIEW

**Comment by PaulTalbot-INL** [comment]
@alfoa @joshua-cogliati-inl @maljovec It looks like the tests are passing on this relatively small but time-sensitive change.

Can anyone look at the merge?

**Comment by joshua-cogliati-inl** [comment]
I'll take a look.

File: framework/utils/randomUtils.py [hunk 1]
```diff
@@ -96,6 +96,7 @@ def randomSeed(value):
     npStochEnv = np.random.RandomState(value)
   else:
     crowStochEnv.seedRandom(value) #this has a void return, right?
+  print('randomUtils: Global random number seed has been changed to',value)
```
Comment (joshua-cogliati-inl) [inline framework/utils/randomUtils.py]: Is this print needed?  Can it be switched to a debug message?

Comment (PaulTalbot-INL) [reply framework/utils/randomUtils.py]: No message handler is available in the utilities, unfortunately.

Comment (joshua-cogliati-inl) [reply framework/utils/randomUtils.py]: Okay, then it is fine as is.

**Comment by PaulTalbot-INL** [comment]
I see that there is no mention of the `samplerInit` for `EnsembleForward`, so I added it.  Thanks.

**Pull request merged**